lib/query/log-warnings.js

```diff
@@ -0,0 +1,65 @@
+const _ = require('lodash')
+const Utils = require('../utils')
+const chalk = require('chalk')
+
+function logWarnings (geojson, format) {
+  const { metadata = {}, features } = geojson
+  const esriFormat = format !== geojson
+
+  if (esriFormat && !metadata.idField) {
+    console.warn(chalk.yellow('WARNING: requested provider has no "idField" assignment. You will get the most reliable behavior from ArcGIS clients if the provider assigns the "idField" to a property that is an unchanging 32-bit integer. Koop will create an OBJECTID field in the absence of an "idField" assignment.'))
+  }
+
+  if (esriFormat && hasMixedCaseObjectIdKey(metadata.idField)) {
+    console.warn(chalk.yellow('WARNING: requested provider\'s "idField" is a mixed-case version of "OBJECTID". This can cause errors in ArcGIS clients.'))
+  }
+
+  // Compare provider metadata fields to feature properties
+  // TODO: refactor
+  if (metadata.fields && _.has(features, '[0].properties')) {
+    warnOnMetadataFieldDiscrepencies(geojson.metadata.fields, geojson.features[0].properties)
```

Comment:
Misspell: `Discrepancies`

I only caught that cause I have a spellcheck extension